mobile_wallet/src/lib.rs

```diff
@@ -1,4 +1,767 @@
+#[macro_use]
+extern crate serde_json;
+use anyhow::{bail, ensure};
+use crypto_common::{
+    types::{Amount, KeyIndex, Signature, TransactionSignature},
+    *,
+};
+use dodis_yampolskiy_prf::secret as prf;
+use ed25519_dalek as ed25519;
+use ed25519_dalek::Signer;
+use either::Either::{Left, Right};
+use encrypted_transfers::encrypt_amount_with_fixed_randomness;
+use id::{account_holder, ffi::AttributeKind, secret_sharing::Threshold, types::*};
+use pairing::bls12_381::{Bls12, G1};
+use rand::thread_rng;
+use serde_json::{from_str, from_value, to_string, Value};
+use sha2::{Digest, Sha256};
+use std::{
+    cmp::max,
+    collections::BTreeMap,
+    convert::TryInto,
+    ffi::{CStr, CString},
+    io::Cursor,
+};
+
+use crypto_common::serde_impls::KeyPairDef;
+type ExampleCurve = G1;
+
+/// Context for a transaction to send.
+#[derive(SerdeDeserialize)]
+#[serde(rename_all = "camelCase")]
+struct TransferContext {
+    pub from:   AccountAddress,
+    pub to:     Option<AccountAddress>,
+    pub expiry: u64,
+    pub nonce:  u64,
+    pub keys:   AccountKeys,
+    pub energy: u64,
+}
+
+/// Sign the given hash.
+fn make_signatures<H: AsRef<[u8]>>(keys: AccountKeys, hash: &H) -> TransactionSignature {
+    // we'll just sign with all the keys we are given, disregarding the threshold.
+    // It is not our job here to decide and in any case the wallet is meant to
+    // support only single key accounts.
+    let mut out = BTreeMap::new();
+    for (cred_index, map) in keys.keys.into_iter() {
+        let mut cred_sigs = BTreeMap::new();
+        for (key_index, kp) in map.keys.into_iter() {
+            let public = kp.public;
+            let secret = kp.secret;
+            let signature = ed25519_dalek::Keypair { public, secret }.sign(hash.as_ref());
+            cred_sigs.insert(key_index, Signature {
+                sig: signature.to_bytes().to_vec(),
+            });
+        }
+        out.insert(cred_index, cred_sigs);
+    }
+    TransactionSignature { signatures: out }
+}
+
+/// Create a JSON encoding of an encrypted transfer transaction.
+fn create_encrypted_transfer_aux(input: &str) -> anyhow::Result<String> {
```

Comment:
Why not add `use anyhow::Result`?

Reply:
It's because Result is an `std` type that is automatically imported, so I think it is confusing to import anyhow::Result unqualified.

And this is how it is in the rest of the project now, so this follows the code style as well.

Reply:
If `anyhow::Result` wasn't an alias for `std::result::Result`, I'd agree it was confusing.

Reply:
Right, but it's not just an alias. It has a default value for the error type. Lots of other crates define their own Result type (e.g., io::Result) which is a similar alias, but with a fixed error type. The error messages you get when using them incorrectly are needlessly confusing when looking at function signatures that look the same. The common way to use such types is to qualify them.